### ISSUE TITLE
[a11y]: Collapsed SideNav Rail aria-hidden is accessibility violation

### ISSUE THREAD

**Issue by JohnWColby** [carbon-design-system/carbon#11122]
### Package

carbon-components-react

### Package version

v7.56.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

When a SideNav Rail is collapsed, `aria-hidden` is set to true, but there are still active buttons within the `nav` element. This is an accessibility violation. (`A focusable element should not be within the subtree of an element with 'aria-hidden' set to "true"`)

This can be seen when running the Accessibility Checker on the Storybook page: https://react.carbondesignsystem.com/?path=/story/components-ui-shell--side-nav-rail

This line should probably take isRail into account: https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/UIShell/SideNav.js#L125

### WCAG 2.1 Violation

1.3.1 Info and Relationships

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-ui-shell--side-nav-rail

### Steps to reproduce

Run Accessibility Checker on SideNav Rail story.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
